### PR TITLE
rustdoc: docs for search deduplication

### PR DESCRIPTION
**Pull request by notriddle**
https://github.com/rust-lang/rust/pull/119912

### CODE REVIEW

**Comment by Urgau** [comment]
The corresponding rustdoc PR was merged. I suppose it's good to proceed?

@notriddle There are some conflicts could you resolve them please.
@GuillaumeGomez Would you be able to review this PR?

File: src/rustdoc-internals/search.md
Comment (GuillaumeGomez) [inline src/rustdoc-internals/search.md:279]: > This might sound a strange design,

Seems a bit strange no?

**Review by GuillaumeGomez — COMMENTED** [review]
Apart from my comment, seems all good to me!

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks all good to me, thanks! I'll let someone with merge rights click on the merge button. :)

**Review by jieyouxu — APPROVED** [review]
Since a rustdoc reviewer said this looks good

**Pull request merged**
